### ISSUE TITLE
Unable to receive Cognito account registration token email - amplify auth with Email Domain Filtering (allowlist) 

### ISSUE THREAD

**Issue by pcstory** [aws-amplify/amplify-cli#7619]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm install -g @aws-amplify/cli

### If applicable, what version of Node.js are you using?

v14.17.0

### Amplify CLI Version

5.0.2

### What operating system are you using?

Mac

### Amplify Categories

auth

### Amplify Commands

Not applicable

### Describe the bug

I am not able to receive Token Email during account registration if I configure " Email Domain Filtering (allowlist)". 

```
amplify add auth 

 The current configured provider is Amazon Cognito.

 Do you want to use the default authentication and security configuration? Default configuration
 Warning: you will not be able to edit these selections.
 How do you want users to be able to sign in? Username
 Do you want to configure advanced settings? Yes, I want to make some additional changes.
 Warning: you will not be able to edit these selections.
 What attributes are required for signing up? Email
 Do you want to enable any of the following capabilities? Email Domain Filtering (allowlist)
? Enter a comma-delimited list of allowed email domains (example: 'mydomain.com, myotherdomain.com'). my-email.com
Successfully added resource mydemoa4b45f38PreSignup locally.
```

### Expected behavior

I should be able to receive email token verification email during account registration.  

### Reproduction steps

1. Follow Step in this page [Setup React Project](https://docs.amplify.aws/start/getting-started/setup/q/integration/react#initialize-a-new-backend.)

2. amplify add auth

```sh
 Do you want to use the default authentication and security configuration? Default configuration
 Warning: you will not be able to edit these selections.
 How do you want users to be able to sign in? Username
 Do you want to configure advanced settings? Yes, I want to make some additional changes.
 Warning: you will not be able to edit these selections.
 What attributes are required for signing up? Email
 Do you want to enable any of the following capabilities? Email Domain Filtering (allowlist)
? Enter a comma-delimited list of allowed email domains (example: 'mydomain.com, myotherdomain.com'). my-domain.com
```

3. Open src/App.js and make the following changes:

```
Import the withAuthenticator component:
import { withAuthenticator } from '@aws-amplify/ui-react'
```

Change the default export to be the withAuthenticator wrapping the main component:
```
export default withAuthenticator(App)
```

4. npm start

5. Perform account registration. 
6. No token email registraction received. 

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by josefaidt** [comment]
Hey @pcstory :wave: thanks for raising this! I was able to reproduce this with the latest version of the CLI using the same auth choices. A few other things I noticed:

- verification emails are sending without the allowlist configuration
- trigger is attached to the Cognito resource, filters appropriately
- removing auth does not remove allowlist pre-signup trigger
- unable to resend codes due to reason "Cannot resend codes. Auto verification not turned on."

Marking this as a bug :)
